metrics.go

```diff
@@ -33,16 +33,16 @@ type Metrics struct {
 	TracesJoinedNotSampled metrics.Counter `metric:"traces" tags:"state=joined,sampled=n"`
 
 	// Number of sampled spans started by this tracer
-	SpansStarted metrics.Counter `metric:"spans" tags:"group=lifecycle,state=started"`
+	SpansStarted metrics.Counter `metric:"spans-by-lifecycle" tags:"state=started"`
```

Comment:
how bout "spans-lifecycle"

Reply:
I'm open to suggestions how to make the naming more consistent across the board.

One other thing I would like to do is to use underscore instead of dash as a separator. Dash is not valid in Prometheus.